### PR TITLE
Add HowardHinnant/date

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **date/2.4.1**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/date/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    def is_sudo_enabled(self):
+        if "CONAN_SYSREQUIRES_SUDO" not in os.environ:
+            if os.name == 'posix' and os.geteuid() == 0:
+                return False
+            if os.name == 'nt':
+                return False
+        return os.getenv("CONAN_SYSREQUIRES_SUDO", True)
+
+    def copy_utc(self):
+        if tools.os_info.is_linux:
```
Comment (SSE4) [inline recipes/date/all/test_package/conanfile.py]: what's very complicated, and you're trying to install something to `/etc` or `/usr`. is there a way to read from the specified file instead?

**Comment by SSE4** [comment]
to be honest, this one concerns me a lot. do I right understand that usage of library depends on deployment of some files into hard-coded locations? and moreover, these locations are only root-writable? and finally, these locations depend from platform to platform?

**Comment by Croydon** [comment]
There we go: We do have two recipes compete for the unpleasantly generic name `date`

#335

**Comment by theodelrieu** [comment]
Actually this is the same library than #335, oops, should have looked at existing PR

**Comment by theodelrieu** [comment]
For the record, the `date` library **ONLY** consists of `date.h`, as Howard Hinnant states here: https://github.com/HowardHinnant/date#summary

I would rather package each library individually. The `sudo` requirements seem to only concern `tz`.

**Comment by SSE4** [comment]
@theodelrieu great, let's package the date.h only then

**Comment by SSE4** [comment]
@uilianries does `date.h` on its own still depend on libcurl?

**Comment by theodelrieu** [comment]
@SSE4 it does not

**Comment by SSE4** [comment]
okay, I am removing the blocking label then
@uilianries 
- pls package date.h only
- change to be fully header-only
- remove libcurl requirement
- remove all root deployment leftover, if any

**Comment by uilianries** [comment]
@SSE4 khorosho

**Comment by uilianries** [comment]
@SSE4 Done! I've updated this recipe to header-only format.

**Comment by SSE4** [comment]
@uilianries bellissimo!

**Pull request merged**